src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -156,11 +160,31 @@ private BoundExpression VisitCollectionInitializerCollectionExpression(BoundColl
         private BoundExpression VisitListInterfaceCollectionExpression(BoundCollectionExpression node)
         {
             Debug.Assert(!_inExpressionLambda);
-            Debug.Assert(node.Type is { });
+            Debug.Assert(node.Type is NamedTypeSymbol);
+
+            var collectionType = (NamedTypeSymbol)node.Type;
+            BoundExpression arrayOrList;
+
+            // Use Array.Empty<T>() rather than List<T> for an empty collection expression when
```

Comment:
I know things are in flux here but I would be fine saying this is just what we do for the empty literal. And the implementation seems fine.